tasks/move-granules/index.js

```diff
@@ -376,38 +350,37 @@ async function moveGranules(event) {
 
   const duplicateHandling = duplicateHandlingType(event);
 
-  const input = get(event, 'input', []);
+  const inputFileList = get(event, 'input', []);
 
   // Get list of cmr file objects from the input Array of S3 filenames (in
   // staging location after processing)
-  const cmrFiles = await getCmrXMLFiles(input, regex);
+  const cmrFiles = await getCmrXMLFiles(inputFileList, regex);
 
   // create granules object for cumulus indexer
-  const allGranules = addInputFilesToGranules(input, inputGranules, regex);
+  const allGranules = addInputFilesToGranules(inputFileList, inputGranules, regex);
 
-  const granulesToMove = updateGranuleMetadata(allGranules, collection, cmrFiles, buckets);
+  const granulesToMove = await updateGranuleMetadata(allGranules, collection, cmrFiles, buckets);
 
   // allows us to disable moving the files
   let movedGranules;
   if (moveStagedFiles) {
     // move files from staging location to final location
-    movedGranules = await moveFilesForAllGranules(granulesToMove, bucket, duplicateHandling);
+    movedGranules = await moveFilesForAllGranules(
+      granulesToMove, bucket, duplicateHandling, buckets
+    );
     // update cmr.xml files with correct online access urls
-    await updateCmrFileAccessURLs(cmrFiles, movedGranules, distEndpoint);
+    await updateEachCmrFileAccessURLs(cmrFiles, movedGranules, distEndpoint, buckets);
+  }
+  else {
+    // TODO [MHS, 2019-01-08] This is the behavior in v1.10.4, but I'm not sure
+    // it's what we want. Validate with someone.  It updates all of the file
+    // location metadata, but doesn't move the files to those locations.  Seems
+    // bad to me.
+    movedGranules = granulesToMove;
```

Comment:
Can someone please verify this?

Reply:
Yeah that sounds wrong. If `moveStagedFiles` is false, we should not update the metadata either, we should just leave it.

Reply:
I'll fix this and add a test to describe the desired behavior.

Reply:
6f79e414b